Branch: feature/2

jurijsmaskaluns opened the pull request:
1. made application refactoring
2. validation fields Name, Category, Description by isEmpty

jurijsmaskaluns committed:
1. made application refactoring
2. validation fields Name, Category, Description by isEmpty

jurijsmaskaluns committed:
Branch: feature/3

task 1 Product name cannot be less than 3 characters and more than 32. ok
task 2 The discount should not be more than 100%. ok
task 3 The discount cannot be less than 0. The discount may be a fractional number (10.5%). ok (discount type is integer)
task 4 The price of the product must be greater than 0. ok
task 7 If the price of the product is less than $ 20, then a discount cannot be made. ok
task 6 The name of the product must be unique. failed

javagururt commented on src/com/javaguru/shoppinglist/repository/ProductInMemoryRepository.java:
```diff
@@ -0,0 +1,28 @@
+    @Override
+    public Product save(Product product) {
+        Product productCopy = new Product(productIdSequence++, product.getName(), product.getPrice(), product.getDiscount(), product.getCategory(), product.getDescription());
+        if(product.getPrice().intValue() <20 ){
```
все же репозиторий за это не отвечает. Он должен просто сохранить. Лучше перенести логику в сервис

javagururt reviewed (commented):
отлично

jurijsmaskaluns committed:
Branch: feature/2

1. add class ProductPriceIsLess (If the price of the product is less than $ 20, then a discount cannot be made.)

jurijsmaskaluns committed:
branch: feature/3
1. build tool Maven
2. add only ProductServiceTest. JUnit + Mockito
